main.cpp

```diff
@@ -0,0 +1,93 @@
+#include <iostream>
```

Comment:
To avoid O(n) iterations over the linked list for append function try to maintain tail pointer as well, so U can append in O(1)